src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -126,53 +133,42 @@ class boss_baron_rivendare : public CreatureScript
             if (!UpdateVictim())
                 return;
 
-            //ShadowBolt
-            if (ShadowBolt_Timer <= diff)
-            {
-                if (SelectTarget(SELECT_TARGET_RANDOM, 0))
-                    DoCastVictim(SPELL_SHADOWBOLT);
-
-                ShadowBolt_Timer = 10000;
-            } else ShadowBolt_Timer -= diff;
-
-            //Cleave
-            if (Cleave_Timer <= diff)
-            {
-                DoCastVictim(SPELL_CLEAVE);
-                //13 seconds until we should cast this again
-                Cleave_Timer = 7000 + (rand32() % 10000);
-            } else Cleave_Timer -= diff;
-
-            //MortalStrike
-            if (MortalStrike_Timer <= diff)
+            _events.Update(diff);
+            while (uint32 eventId = _events.ExecuteEvent())
             {
-                DoCastVictim(SPELL_MORTALSTRIKE);
-                MortalStrike_Timer = 10000 + (rand32() % 15000);
-            } else MortalStrike_Timer -= diff;
-
-            //RaiseDead
-            //            if (RaiseDead_Timer <= diff)
-            //          {
-            //      DoCast(me, SPELL_RAISEDEAD);
-            //                RaiseDead_Timer = 45000;
-            //            } else RaiseDead_Timer -= diff;
-
-            //SummonSkeletons
-            if (SummonSkeletons_Timer <= diff)
-            {
-                me->SummonCreature(11197, ADD_POS_1, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_2, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_3, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_4, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_5, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_6, TEMPSUMMON_TIMED_DESPAWN, 29000);
-
-                //34 seconds until we should cast this again
-                SummonSkeletons_Timer = 40000;
-            } else SummonSkeletons_Timer -= diff;
+                switch (eventId)
+                {
+                    case EVENT_SHADOW_BOLT:
+                        if (SelectTarget(SELECT_TARGET_RANDOM, 0))
+                            DoCastVictim(SPELL_SHADOWBOLT);
+                        _events.ScheduleEvent(EVENT_SHADOW_BOLT, 10 * IN_MILLISECONDS);
+                        break;
+                    case EVENT_CLEAVE:
+                        DoCastVictim(SPELL_CLEAVE);
+                        _events.ScheduleEvent(EVENT_CLEAVE, urand(7, 17) * IN_MILLISECONDS);
+                        break;
+                    case EVENT_MORTAL_STRIKE:
+                        DoCastVictim(SPELL_MORTALSTRIKE);
+                        _events.ScheduleEvent(EVENT_MORTAL_STRIKE, urand(10, 25) * IN_MILLISECONDS);
+                        break;
+                        /*case EVENT_RAISE_DEAD:
```

Comment:
The reason this doesnt work is this - SPELL_RAISEDEAD = 15708 // bad copypaste of value (15708 is mortal strike)